### PR TITLE
Build admin 28 may

### PR DESCRIPTION
**Pull request by rohit-walkover**

### CODE REVIEW

File: apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts [hunk 1]
```diff
@@ -74,6 +91,15 @@ export class LicenceKeyComponent implements OnInit {
         this.getAllLicenceKey();
     }
 
+    /**
+     *To hide column filter pop up
+     *
+     * @memberof LicenceKeyComponent
+     */
+    public hideListItems() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts:99]: add return type

File: apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts [hunk 2]
```diff
@@ -104,4 +130,134 @@ export class LicenceKeyComponent implements OnInit {
...
+        });
+
+    }
+
+    /**
+    * This function is used to reset filters
+    *
+    * @memberof LicenceKeyComponent
+    */
+    public resetFilters() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts:160]: return type

File: apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts [hunk 3]
```diff
@@ -104,4 +130,134 @@ export class LicenceKeyComponent implements OnInit {
...
+    }
+
+    /**
+     * To apply column toggle filter
+     *
+     * @param {boolean} event boolean is column show or hide
+     * @param {string} column Column name
+     * @memberof LicenceKeyComponent
+     */
+    public columnFilter(event: boolean, column: string) {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts:210]: return type

File: apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts [hunk 4]
```diff
@@ -104,4 +130,134 @@ export class LicenceKeyComponent implements OnInit {
...
+        this.isAllColumnFilterApplied()
+        this.updateColumnFilter();
+    }
+
+    /**
+    *To check all column filter applied true
+    *
+    * @memberof LicenceKeyComponent
+    */
+    public isAllColumnFilterApplied() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts:245]: return type

File: apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts [hunk 5]
```diff
@@ -104,4 +130,134 @@ export class LicenceKeyComponent implements OnInit {
...
+        this.isAllFieldColumnFilterApplied = Object.keys(this.showFieldFilter).every((k) => this.showFieldFilter[k]);
+    }
+
+
+    /**
+     * To get count of colspan
+     *
+     * @memberof LicenceKeyComponent
+     */
+    public getColspanCount() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/licence-key/licence-key-component/licence-key.component.ts:255]: return type

File: apps/web-giddh-admin/src/app/models/company.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    signUpOn: boolean;
+    status: boolean;
+    subscribedOn: boolean;
+    subscriptionId: boolean;
+    totalAmount: boolean;
+    transactionLimit: boolean;
+
+}
+
+export class PlanFieldFilterColumnNames {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/models/company.ts:43]: run formatter

File: apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts [hunk 7]
```diff
@@ -137,6 +144,15 @@ export class PlansComponent implements OnInit {
         this.getAllPlans();
     }
 
+    /**
+     * To hide column filter pop up
+     *
+     * @memberof PlansComponent
+     */
+    public hideListItems() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts:152]: return type

File: apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts [hunk 8]
```diff
@@ -346,6 +368,130 @@ export class PlansComponent implements OnInit {
...
+    }
+
+    /**
+    *To apply column toggle filter
+    *
+    * @param {boolean} event boolean is column show or hide
+    * @param {string} column Column name
+    * @memberof PlansComponent
+    */
+    public columnFilter(event: boolean, column: string) {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts:450]: return type

File: apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts [hunk 9]
```diff
@@ -346,6 +368,130 @@ export class PlansComponent implements OnInit {
...
+        this.isAllColumnFilterApplied();
+        this.updateColumnFilter();
+    }
+
+      /**
+     *To check all column filter applied true
+     *
+     * @memberof PlansComponent
+     */
+    public isAllColumnFilterApplied() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts:479]: return type

File: apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts [hunk 10]
```diff
@@ -346,6 +368,130 @@ export class PlansComponent implements OnInit {
...
+        this.isAllFieldColumnFilterApplied = Object.keys(this.showFieldFilter).every((k) => this.showFieldFilter[k]);
+    }
+    
+
+    /**
+    * To get count of colspan
+    *
+    * @memberof PlansComponent
+    */
+    public getColspanCount() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/plans/plans-component/plans.component.ts:489]: return type

File: apps/web-giddh-admin/src/app/services/column-filter.service.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+        private http: HttpWrapperService,
+        @Optional() @Inject(ServiceConfig) private config: IServiceConfigArgs) {
+    }
+
+    /**
+     * API call to get favorite page column 
+     * 
+     * @param {string} pageName Module/Page name 
+     */
+    public getFavouritePage(pageName: string) {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/services/column-filter.service.ts:22]: return type

File: apps/web-giddh-admin/src/app/services/column-filter.service.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+    /**
+     * API call to update favorite column of an module which is selected by user
+     *
+     * @param {string} pageName Module/Page name 
+     * @param {*} post Request body
+     * @returns
+     * @memberof UserService
+     */
+    public updateFavouritePage(pageName: string, post: any) {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/services/column-filter.service.ts:43]: return type

File: apps/web-giddh-admin/src/app/services/subscription.service.ts [hunk 13]
```diff
@@ -88,6 +88,47 @@ export class SubscriptionService {
...
     }
 
+    /**
+     * To get company footer data
+     *
+     * @param {GetAllCompaniesRequest} body
+     * @returns
+     * @memberof SubscriptionService
+     */
+    public getCompaniesFooter(body: GetAllCompaniesRequest) {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/services/subscription.service.ts:98]: return type

File: apps/web-giddh-admin/src/app/services/subscription.service.ts [hunk 14]
```diff
@@ -88,6 +88,47 @@ export class SubscriptionService {
...
+    }
+
+    /**
+    * To get company permission
+    *
+    * @param {string} companyUniqueName company unique name
+    * @returns
+    * @memberof SubscriptionService
+    */
+    public getCompaniesPermission(companyUniqueName: string) {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/services/subscription.service.ts:116]: return type

File: apps/web-giddh-admin/src/app/services/user.service.ts [hunk 15]
```diff
@@ -82,4 +82,26 @@ export class UserService {
...
     }
+
+    /**
+     * This will export the subscription by users
+     *
+     * @param {*} model
+     * @returns
+     * @memberof UserService
+     */
+    public exportSubscriptionsByUsers(model: CommonPaginatedRequest, post: any) {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/services/user.service.ts:93]: return type

File: apps/web-giddh-admin/src/app/subscription/components/company-advance-search/company-advance-search.component.ts [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+        transactionLimit: '',
+        additionalChargesOperation: '',
+        additionalCharges: ''
+    };
+    public lessGreaterFilters: IOption[] = [{ label: 'Greater than', value: 'GREATER_THAN' }, { label: 'Less than', value: 'LESS_THAN' }, { label: 'Greater than equals to', value: 'GREATER_THAN_OR_EQUALS' }, { label: 'Less than equals to', value: 'LESS_THAN_OR_EQUALS' }];
+    public defaultLoad: boolean = true;
+
+    constructor(private fb: FormBuilder, private store: Store<AppState>, private adminActions: AdminActions, private toasty: ToasterService) { }
+
+    ngOnInit() {
```
Comment (ravinderwalkover) [inline apps/web-giddh-admin/src/app/subscription/components/company-advance-search/company-advance-search.component.ts:64]: return type in all functions in this PR

Comment (rohit-walkover) [reply apps/web-giddh-admin/src/app/subscription/components/company-advance-search/company-advance-search.component.ts:64]: it was existing code

File: apps/web-giddh-admin/src/app/services/user.service.ts (hunk 15 above)
Comment (rohit-walkover) [reply apps/web-giddh-admin/src/app/services/user.service.ts:93]: have not work on it i think you have worked on it

**Pull request merged**